### ISSUE TITLE
Allow selecting multiple genders for 

### ISSUE THREAD

**Issue by mnalis** [openstreetmap/iD#9591]
### Description

Currently, for [shop=hairdresser](https://wiki.openstreetmap.org/wiki/Tag:shop=hairdresser), iD allows selecting **only one** of the gender values [male=*](https://wiki.openstreetmap.org/wiki/Key:male), [female=*](https://wiki.openstreetmap.org/wiki/Key:female) and [unisex=*](https://wiki.openstreetmap.org/wiki/Key:unisex), i.e. UI control is radio-button.

That however presents a problem when users try to edit objects that have more than one value set - most commonly combination of `male` + `female` tags, especially due to `unisex=*` ambiguity/controversy documented on its [wiki page](https://wiki.openstreetmap.org/wiki/Key:unisex) and related iD issues (like related https://github.com/openstreetmap/iD/issues/8340 and https://github.com/openstreetmap/iD/issues/7427).

Request: It would be preferable if iD allowed entering all three gender keys (`male`, `female`, `unisex`) via  **checkboxes instead of radiobuttons** 

This is how most other popular editors do it (screenshots of JOSM and Vespucci attached). Current iD handling of that situation is very subtle, and unless user has display of raw tags displayed and uses that instead of user-friendly presets, is likely to result in _unintended_ modification / retagging / data loss  on edits. (see iD picture, in which iD would_ silently remove all gender tags except one that was selected via radio button_ if it were clicked on, regardless of their previous combinations/values)

### Screenshots

![josm](https://user-images.githubusercontent.com/156656/233216279-7679b2c8-c7e5-4445-bb57-a5481eba4a67.png)
![small_Screenshot_20230420_004438_de blau android](https://user-images.githubusercontent.com/156656/233216297-21262ef7-e313-4fec-8078-e3db8b391dcc.jpg)
![id](https://user-images.githubusercontent.com/156656/233217152-2ed8a7a6-5238-4a2e-b4d4-94abb2b8b7fd.png)

**Comment by westnordost** [comment]
So both the original meaning of the tag and the meaning of "unisex" used by the common native speaker for restrooms is a restroom without segregation by sex. So `unisex=yes` implies that the restroom can be used both by men and women, i.e. it is not segregated by sex. The logical fallacy made in the design of this select field in iD was that the reverse does not apply: If a restroom can be used both by men and women, it does **not** necessarily mean that it is a unisex restroom.

So, the implication above is for data consumers to make, not for editor software. i.e. data consumers wanting to find all restrooms that can be used by women will want to filter for `unisex=yes or female=yes`.

If I remember correctly, the current UI decision was rationalized by previous iD maintainers saying that ideally, there would be one `amenity=toilet` node for every sex when it is a segregated restroom. But this conflicts with reality, de-facto, this is not how restrooms are mapped in most cases.

**Comment by 1ec5** [comment]
> If I remember correctly, the current UI decision was rationalized by previous iD maintainers saying that ideally, there would be one `amenity=toilet` node for every sex when it is a segregated restroom. But this conflicts with reality, de-facto, this is not how restrooms are mapped in most cases.

There’s something to be said for this approach, but there’s nothing in the UI to guide the user to this realization, unlike with, say, sidewalks, in which the path of least resistance is currently to map a separate way. Not every native English speaker has a correct understanding of the term “unisex”, let alone non-native speakers.

But if I’m reading this ticket correctly, it’s actually about `shop=hairdresser` rather than `amenity=toilets`. Apparently the same [`gender` field](https://github.com/openstreetmap/id-tagging-schema/blob/723c87595b668a99d9c5faaefc50d1984a580250/data/fields/gender.json) has been reused for both presets, as well as for [`amenity=shower`](https://github.com/openstreetmap/id-tagging-schema/blob/65f46b5adb7252c1af2875129be26becea1eed25/data/presets/amenity/shower.json#L16) and [`amenity=dressing_room`](https://github.com/openstreetmap/id-tagging-schema/blob/65f46b5adb7252c1af2875129be26becea1eed25/data/presets/amenity/dressing_room.json#L6). For some of these presets, there’s only one possible interpretation of the “Unisex” option, and it isn’t the one that would require each gender’s facility to be mapped separately.

**Comment by mnalis** [comment]
- for the topic of toilets; see the related issues https://github.com/openstreetmap/iD/issues/8340 and https://github.com/openstreetmap/iD/issues/7427 (where I agree that the same checkbox solution could be applied).
- this issue is primarily (quite related, but still somewhat different, as noted by @1ec5) for `shop=hairdresser`, outlining the issue that **users do currently map multiple** of those tags (`male`/`female`/`unisex`) **at the same time**, and that iD current UI is conductive to unsuspectingly  damaging such tagging (while noting that other popular editors employ checkboxes to avoid such issue, so iD could employ that tactics too). 
- whether such multi-key tagging is the best way to tag what hairstyles/services are offered, or if some other method would be preferable (like single-key `unisex=yes` or [for=](https://wiki.openstreetmap.org/wiki/Key:for)=`woman` would be better etc.)  was intentionally **beyond the scope of this issue** (which is, again, primarily to **protect against the damage to existing tagging methods**)

**Comment by JesseWeinstein** [comment]
This might be better opened on the https://github.com/openstreetmap/id-tagging-schema repo, as the (one-line) change is to solve it is: https://github.com/openstreetmap/id-tagging-schema/blob/main/data/fields/gender.json#L2 "type" field from "radio" to "semiCombo".

**Comment by westnordost** [comment]
<sub>FYI an issue can be transferred within the same organization using this link on the right, let's leave it up to the maintainer whether or not it should be transferred to the other repo
![image](https://user-images.githubusercontent.com/4661658/235296875-3d657212-d7d5-4777-90ed-cfff7171fb81.png)
</sub>

**Comment by mnalis** [comment]
Thanks for the hint @JesseWeinstein , but are you sure just changing that  `type` from `radio` to `semiCombo` would work? Do you know of **existing tagging that works in that way?** (i.e.with multiple keys, instead of multiple values for one key separated by semicolons)

I've tried looking, but it seems that all occurrences I found that use `semiCombo` are **one key** + multiple values. (and here we have **multiple keys** `male=*`/`female=*`/`unisex=*` instead), which ends up looking/working like this (e.g. [beauty.json](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/fields/beauty.json#L3):

![iD_beauty](https://user-images.githubusercontent.com/156656/235302791-cdba4246-c6eb-433c-aaed-1389413d8ae4.png)

And that doesn't seem like it would accomplish what we need for `shop=hairdresser`?

**Comment by bhousel** [comment]
> I've tried looking, but it seems that all occurrences I found that use `semiCombo` are **one key** + multiple values. (and here we have **multiple keys** `male=*`/`female=*`/`unisex=*` instead), which ends up looking/working like this

yes!  #3954 is the issue that everyone here needs to read.  Switching from a radio to a semiCombo won't fix the problem.

The accepted tagging for gendered things does not work like other multi valued tags in OSM.

Solution is either - need to wait for better tags to be proposed (which was where #3954 left things), or replace the existing radiobutton gender field with a bunch of gender checkbox fields, or develop a new kind of special field that can manage all the different tags.

**Comment by mnalis** [comment]
Thanks @bhousel! That is how it seemed to me too, that one would need 3 separate checkboxes (`male`, `female`, `unisex`) which are tri-state (undef, `yes`, `no`). So it would look something like those tags:

![id-checkbox-hairdresser-example](https://user-images.githubusercontent.com/156656/235307303-ac00efc3-98a9-449e-89bc-0c4c7ca66790.png)

Of course, it would look ***nicer*** if those 3 separate checkboxes could be grouped somehow together, but that would be just visual fluff (i.e. functionality would remain the same; but it would use less screen and look nicer). However, I wasn't able to find a tagging that works that way, and you confirm that such nicer layout not exist yet, so one would currently be fixed on using existing `checkbox` functionality like in screenshot above. 

As an aside, disadvantage of UI like for `recycling_type=container`  (mentioned in https://github.com/openstreetmap/iD/issues/3954#issuecomment-293019220) is that it **does not distinguish** between unsurveyed (not tagged) case (e.g. missing `male=*` tag), and explicitly tagged `no` case (e.g. `male=no`). Which works fine for recycling containers (because their [wiki](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Drecycling) explicitly declares that non mentioned tag defaults to `no`) but does not really work for tags where it is not explicitly implied (like `male`, `fuel:*`, `payment:*` etc) and where in fact explicit `female=no` is even more important than explicit `female=yes` (not to mention untagged/unknown value when tag is missing altogether)
